### ISSUE TITLE
Oversized icon in tray applet device settings

### ISSUE THREAD

**Issue by mdinslage** [polychromatic/polychromatic#348]
Hello,
I have been making Slackware packages for openrazer, razergenie, razercommander, and polychromatic that I will be submitting to www.slackbuilds.org when Slackware 15 gets released in the coming weeks.   Everything is working fine, but today I noticed a visual bug.  I installed openrazer and polychromatic on my laptop to configure my Razer Atheris mouse.  Everything is working great, but I found that when I tried to adjust settings for the device under the tray applet the sleep icon is ridiculously oversized.  I am thinking this might be a bug.  Attached is a screenshot of the problem.  

![razer_poly](https://user-images.githubusercontent.com/28424337/129969082-587480f3-12e3-41e4-b5bf-ea3fa849b522.png)

If this is something I can fix on my end please let me know.

Thanks in advance.

**Comment by lah7** [comment]
Hi, thank you for packaging for Slackware users! :+1: 🎁 

The tray applet supports one of two technologies -- what you're seeing is the "fallback" GTK Status Icon which doesn't attempt any resizing of the original SVGs, so that would be a bug on this side. Technically, GTK deprecated this too.

The other is `AppIndicator3` (GObject bindings) which would be a better option to package if it's available for your distro. If it's any help, [this is the dependency in Debian (gir1.2-appindicator3-0.1)](https://packages.debian.org/sid/gir1.2-appindicator3-0.1).

I'm going to look into porting the tray applet to QSystemTrayIcon as a third option too, to remove the dependency on GTK.

**Comment by mdinslage** [comment]
Ah makes sense, I did see that message when launching the tray applet from the mate-terminal.   AppIndicator3 is not in the default install of Slackware, but it is available at www.slackbuilds.org.  I will just make a note in my package README that AppIndicator3 is a optional dependency.   I will go ahead and test it with AppIndicator3.

**Comment by mdinslage** [comment]
Ok so I was thinking of Slackware 14.2 when I said AppIndicator3 is not in the default Slackware install.  A few months back it was added to Slackware development branch which I am currently running.  https://slackware.osuosl.org/slackware64-current/source/l/libappindicator/

This is the error I get when launching...

```
/usr/bin/polychromatic-tray-applet:200: DeprecationWarning: Gtk.ImageMenuItem.set_image is deprecated
  item.set_image(img)            # GTK3: Deprecated, no replacement!
```
I am not sure but I assume this means that its not using libappindicator but falling back to GTK Status Icon?

**Comment by mdinslage** [comment]
After looking over the Slackware build scripts for libappindicator and its dependencies it does build both gtk2 and gtk3 libraries.  Slackware should have what it needs for using libappindicator.  but I decided to do a workaround, I just resampled the idle_time.svg icon down to 24x24 in my buildscript and that fixes the problem.   I only have a few razer devices to test with, but looking at all the icons they all seem to be between 22x22 and 48x48 size.   If I need to I can always have my script loop through all the the svg's and resample them to the same size.   but so far on two computers with different devices the icons look fine other than the idle_time.svg which I now fixed.  
![razer2](https://user-images.githubusercontent.com/28424337/129983588-ebb50965-f853-4065-aeab-94365ab81749.png)

**Comment by mdinslage** [comment]
Last post promise :).  So I realized that I am running mate-desktop which is not an official part of Slackware.   I tested the tray-applet in KDE5 and XFce which are both officially part of Slackware.  On both DE's the tray icons scaled correctly.  Not sure if what the issue is with Mate but I am not going to worry about it.   I will close this now.

**Comment by lah7** [comment]
It's possibly expected on MATE with its default configuration. The same happens on Debian, but not under Ubuntu MATE. I'm aware Ubuntu MATE configure their defaults to use indicators (since 20.10, using [Ayatana Indicators](https://ayatanaindicators.github.io/)). Perhaps AppIndicator3 fallbacks anyway if indicators are not configured for an environment?

Resizing seems like a good solution. For icons that are shared between the tray and Controller, I'm just aware from having a HiDPI display that Qt doesn't upscale smaller sized SVGs that well, which is why I left them slightly larger or at original sizes. I'll have to check again. I don't think that icon is used in the Controller, so it's all good. 

The green colours (#344) is something I'd also like to fix, but GTK is crippling here! Likewise with the deprecation, GTK wants to get rid of icons in menus. 🤷🏼‍♂️ 

----

I'd like to add Slackware to the website's download page as a community packaged option once it's available. Please could you let me know instructions on how a user could install it and a URL to the package?

**Comment by mdinslage** [comment]
Yeah I did some digging and saw that the mate packages for Slackware did not include mate-indicator-applet.  I wasn't sure if this was something I needed or not, but installing it did not fix the issue.  Since mate is not an official DE in the stock Slackware I just edited my buildscript to run a check for mate and if it finds it then it will resize all the svg icons to 24x24.  Otherwise it will just leave them as is since this issue doesn't effect KDE5 or XFce.   So far just testing on my laptop mouse and desktop mouse everything looks fine in both the tray applet and controller.  I was using razergenie for a while, but was finally able to get 0.7.0 to compile on the latest Slackware and its a great piece of Software.

And as soon as my builds are submitted and approved I will let you know so you can add them to your websites download page.  Slackware 15 just hit release candidate 1 status, so it will probably go stable release in a month or so.  Right now www.slackbuilds.org is not accepting new submissions so they can get ready for the next release.   Once stable is released they will reopen up again and I can get them submitted for approval.
![razer33](https://user-images.githubusercontent.com/28424337/130046131-168334d7-45b3-4f2c-80fd-614a76a4d74c.png)

**Comment by lah7** [comment]
I'm going to reopen this issue to track the oversized icon bug. Even if GTK deprecated the Status Icon, it might as well work properly.

I can probably tackle this at the same time as detecting a suitable icon colour.
